subliminal/providers/itasa.py

```diff
@@ -0,0 +1,390 @@
+            return []
+
+        # get the page of the season of the show
+        logger.info('Getting the subtitle of show id %d, season %d episode %d, format %s', show_id,
+                    season, episode, format)
+        subtitles = []
+
+        params = {
+            'apikey': self.apikey,
+            'show_id': show_id,
+            'q': '%dx%02d' % (season, episode),
+            'version': self.itasa_formats.get(format.lower() if format is not None else 'hdtv')
+            # Default format is HDTV
+            }
+        logger.debug(params)
+        r = self.session.get(self.server_url + 'subtitles/search', params=params, timeout=30)
+        r.raise_for_status()
+        root = ParserBeautifulSoup(r.content, ['lxml-xml', 'lxml', 'html.parser']).find('root')
+
+        if int(root.data.count.text) == 0:
+            logger.warning('Subtitles not found')
+            return []
+
+        # Looking for subtitlles in first page
+        for subtitle in root.data.subtitles:
+            if '%dx%02d' % (season, episode) in subtitle.find('name').text.lower():
+
+                logger.debug('Found subtitle id %d - %s - %s',
+                             int(subtitle.id.text),
+                             subtitle.find('name').text,
+                             subtitle.version.text)
+
+                sub = ItaSASubtitle(
+                        int(subtitle.id.text),
+                        subtitle.show_name.text,
+                        season,
+                        episode,
+                        format,
+                        subtitle.find('name').text)
+
+                subtitles.append(sub)
+
+        # Not in the first page of result try next (if any)
+        next = root.find('next')
+        while next.text != '':
+
+            r = self.session.get(next.text, timeout=30)
+            r.raise_for_status()
+            root = ParserBeautifulSoup(r.content, ['lxml-xml', 'lxml', 'html.parser']).find('root')
+
+            logger.info('Loading subtitles page %s', root.data.page.text)
+
+            # Looking for show in following pages
+            for subtitle in root.data.subtitles:
+                if '%dx%02d' % (season, episode) in subtitle.find('name').text.lower():
+
+                    logger.debug('Found subtitle id %d - %s - %s',
+                                 int(subtitle.id.text),
+                                 subtitle.find('name').text,
+                                 subtitle.version.text)
+
+                    sub = ItaSASubtitle(
+                        int(subtitle.id.text),
+                        subtitle.show_name.text,
+                        season,
+                        episode,
+                        subtitles.version.text,
+                        subtitle.find('name').text)
+
+                    subtitles.append(sub)
+
+            next = root.find('next')
+
+        # Dowload the subs found, can be more than one in zip
```

Comment:
Why download the subtitles? This should be done in the `download` method instead.